lib/srv/db/postgres/users.go

```diff
@@ -0,0 +1,159 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package postgres
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"strings"
+
+	"github.com/gravitational/teleport/lib/srv/db/common"
+
+	"github.com/gravitational/trace"
+	"github.com/jackc/pgx/v4"
+)
+
+// ActivateUser creates or enables the database user.
+func (e *Engine) ActivateUser(ctx context.Context, sessionCtx *common.Session) error {
+	if sessionCtx.Database.GetAdminUser() == "" {
+		return trace.BadParameter("admin user is not configured for this database, cannot activate the user")
+	}
+
+	conn, err := e.pgxConnect(ctx, sessionCtx.WithUser(sessionCtx.Database.GetAdminUser()))
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	defer conn.Close(ctx)
+
+	// We could call this once when the database is being initialized but
+	// doing it here has a nice "self-healing" property in case the Teleport
+	// bookkeeping group or stored procedures get deleted or changed offband.
+	err = e.initAutoUsers(ctx, conn)
```

Comment:
Database agent is usually close to the database server but three calls may add up. Do we have a rough estimate of how much latency this adds?

Reply:
Yeah, well, we're already adding basically another connection roundtrip for each auto-provisioned user, right? I don't think added latency for 3 extra queries here is huge but we can see if it becomes a problem and optimize then?